packages/validation/src/utils/joi-to-error-extensions.ts

```diff
@@ -94,6 +94,27 @@ export const joiValidationErrorItemToErrorExtensions = (
 		extensions.invalid = validationErrorItem.context?.value;
 	}
 
+	// TODO Find a better way of passing the expected value down to the client
```

Comment:
regarding this, I wonder whether it is possible to do something similar to the existing `contains` and `ncontains` which are an extension of the Joi instance:

https://github.com/directus/directus/blob/b211a3b68d4af52237decc68fc556941fb5935ec/packages/utils/shared/generate-joi.ts#L18-L59

which may make it possible to attach the "substring" even for all variations of starts_with/ends_with without having to extract it from `validationErrorItem.context?.['regex']`. I'm still not 100% sure why/how `contains` and `ncontains` returns "substring" in the context (unless "substring" being in `args` is the factor here)

however that can/should be a separate PR, so this particular review comment shouldn't block this particular PR.

Reply:
Took a quick look at this. Could work, but the extension would need to be named differently (cannot override `pattern`). Then we would have to replace usages of `pattern` with this extension. Hopefully we could just reuse `pattern` in this extension instead of re-implementing the whole logic.
In summary, a bit much effort just to get the value easier.